### PR TITLE
Feat: Add Support for Italian 

### PR DESCRIPTION
**Pull request by viggo-gascou**
This PR aims to add support for evaluating models on Italian, by adding the following datasets (partly from #755):

- [x] Question answering: [SQuAD-it](https://huggingface.co/datasets/crux82/squad_it)
- [x]     Summarization: [IlPost](https://huggingface.co/datasets/ARTeLab/ilpost)
- [x]     Sentiment classification: [SentiPolc-16](https://huggingface.co/datasets/cardiffnlp/tweet_sentiment_multilingual/viewer/italian)
- [x]     Reasoning: [HellaSwag-it](https://huggingface.co/datasets/alexandrainst/m_hellaswag/viewer/it)
- [x]     Named entity recognition: [wikineural](https://huggingface.co/datasets/Babelscape/wikineural) and [multinerd](https://huggingface.co/datasets/Babelscape/multinerd)
- [x]     Linguistic acceptability: ScaLA algorithm with [Italian Universal Dependencies](https://github.com/UniversalDependencies/UD_Italian-ISDT)
- [x]     Knowledge: [MMLU-it](https://huggingface.co/datasets/alexandrainst/m_mmlu/viewer/it)

### CODE REVIEW

File: docs/datasets/italian.md [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+
+
+## Reading Comprehension
+
+### SQuAD-it
+
+This dataset is derived from the SQuAD 1.1 dataset and was published in
+[this paper](https://link.springer.com/chapter/10.1007/978-3-030-03840-3_29).
+The questions and answers were obtained through "semi-automatic" translation of the
+SQuAD dataset to Italian. The dataset consists of 54,159 / 7,609 question/answer pairs
```
Comment (saattrupdan) [inline docs/datasets/italian.md]: Can you mention what translation model was used here?

File: docs/datasets/italian.md [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+  ```
+  Articolo di cronaca: {text}
+
+  Scrivete un riassunto dell'articolo sopra citato.
+  ```
+
+You can evaluate this dataset directly as follows:
+
+```bash
+$ scandeval --model <model-id> --dataset ilpost-sum
```
Comment (saattrupdan) [inline docs/datasets/italian.md]: Is it `ilpost` or `ilpost-sum`? It should be the same in the command as in the heading.

**Review by saattrupdan — CHANGES_REQUESTED** [review]
Looks good so far! A few smaller comments, but otherwise perfect 🙂

File: docs/datasets/italian.md (hunk 1 above)
Comment (viggo-gascou) [reply docs/datasets/italian.md]: It seems to be DeepL, I'll add that! It also seems that they did inspect the translations and discarded any that did not contain the answer, of which most were actually incorrect translations.

File: docs/datasets/italian.md (hunk 2 above)
Comment (viggo-gascou) [reply docs/datasets/italian.md]: I think ilpost-sum makes more sense, that way it is more clear that it is a summarization dataset, I'll fix it!

File: docs/datasets/italian.md [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+```
+
+
+## Named Entity Recognition
+
+### MultiNERD IT
+
+This dataset was published in [this
+paper](https://aclanthology.org/2022.findings-naacl.60/) and
+consists of sentences from Wikipedia and Wikinews in 10 different languages. It an
```
Comment (saattrupdan) [inline docs/datasets/italian.md]:
```suggestion
consists of sentences from Wikipedia and Wikinews in 10 different languages. It is an
```

File: docs/datasets/italian.md [hunk 4]
```diff
@@ -0,0 +1,575 @@
...
+- `MYTH` ➡️ `MISC`
+- `PLANT` ➡️ `MISC`
+- `VEHI` ➡️ `MISC`
+
+Here are a few examples from the training split:
+
+```json
+{
+  "tokens": "['Alcune' 'statue' 'che' 'la' 'rappresentano' 'vennero' 'ritrovate' 'non' 'lontano' 'da' 'Tani' ',' 'anche' 'se' 'in' 'nessuna' 'di' 'queste' 'si' 'è' 'conservato' 'il' 'volto' ',' 'mentre' 'nella' 'seconda' 'cateratta' 'è' 'registrata' 'una' 'piena' 'del' 'Nilo' 'datata' 'al' 'suo' '3º' 'anno' 'di' 'regno' '.']",
+  "labels": "['O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'B-LOC', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'B-LOC', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O']"
```
Comment (saattrupdan) [inline docs/datasets/italian.md]: This doesn't match the other NER datasets, which are stored as arrays instead of JSON strings. Should this be changed?

File: docs/datasets/italian.md [hunk 5]
```diff
@@ -0,0 +1,575 @@
...
+consists of sentences from Wikipedia and Wikinews in 10 different languages. It an
+extension of the combination of
+(WikiNEuRal)[https://www.github.com/Babelscape/wikineural] and
+(NER4EL)[https://www.github.com/Babelscape/ner4el]. The original test set was created
+from manual annotations, while the training set is based on an automatic annotation
+pipeline.
+
+The Italian part of the original dataset consists of 181,000 sentences. We use the given
+training, validation and testing sets, and use 1,024 / 256 / 2,048 samples for training,
+validation, and testing, respectively.
```
Comment (saattrupdan) [inline docs/datasets/italian.md]: Was the original dataset already split into train/val/test? If so, are our new splits subsets of the original?

File: docs/datasets/italian.md [hunk 6]
```diff
@@ -0,0 +1,575 @@
...
+### Unofficial: WikiNEuRal IT
+
+This dataset was published in [this
+paper](https://aclanthology.org/2021.findings-emnlp.215) and
+consists of sentences from Wikipedia in 9 different languages. The annotations are
+automatic but at the time novel and state-of-the-art methodologies.
+
+The Italian part of the original dataset consists of 111,000 sentences. We use the given
+training, validation and testing sets, and use 1,024 / 256 / 2,048 samples for training,
+validation, and testing, respectively.
```
Comment (saattrupdan) [inline docs/datasets/italian.md]: Was the original dataset already split into train/val/test? If so, are our new splits subsets of the original?

File: docs/datasets/italian.md [hunk 7]
```diff
@@ -0,0 +1,575 @@
...
+$ euroeval --model <model-id> --dataset scala-it
+```
+
+
+## Reading Comprehension
+
+### SQuAD-it
+
+This dataset is derived from the SQuAD 1.1 dataset and was published in
+[this paper](https://link.springer.com/chapter/10.1007/978-3-030-03840-3_29).
```
Comment (saattrupdan) [inline docs/datasets/italian.md]: DOI's are generally preferred, as they are more future proof:

```suggestion
[this paper](https://doi.org/10.1007/978-3-030-03840-3_29).
```

File: src/euroeval/dataset_configs.py [hunk 8]
```diff
@@ -718,6 +738,85 @@ def get_dataset_config(dataset_name: str) -> DatasetConfig:
     unofficial=True,
 )
 
+WIKINEURAL_IT_CONFIG = DatasetConfig(
+    name="wikineural-it",
+    pretty_name="the truncated version of the multilingual named "
```
Comment (saattrupdan) [inline src/euroeval/dataset_configs.py]:
```suggestion
    pretty_name="the truncated version of the Italian named "
```

File: src/euroeval/dataset_configs.py [hunk 9]
```diff
@@ -718,6 +738,85 @@ def get_dataset_config(dataset_name: str) -> DatasetConfig:
...
+    "'posizione', 'organizzazione' e 'varie'. I valori devono essere elenchi di entità "
+    "nominate di quel tipo, esattamente come appaiono nella frase.",
+    num_few_shot_examples=8,
+    max_generated_tokens=128,
+    unofficial=True,
+)
+
+MULTINERD_IT_CONFIG = DatasetConfig(
+    name="multinerd-it",
+    pretty_name="the truncated version and Italian part of the multilingual named "
```
Comment (saattrupdan) [inline src/euroeval/dataset_configs.py]:
```suggestion
    pretty_name="the truncated version of the Italian part of the named "
```

File: src/euroeval/dataset_configs.py [hunk 10]
```diff
@@ -1580,6 +1728,23 @@ def get_dataset_config(dataset_name: str) -> DatasetConfig:
     max_generated_tokens=5,
 )
 
+MMLU_IT_CONFIG = DatasetConfig(
+    name="mmlu-it",
+    pretty_name="the truncated version of the French knowledge dataset MMLU-it, "
```
Comment (saattrupdan) [inline src/euroeval/dataset_configs.py]:
```suggestion
    pretty_name="the truncated version of the Italian knowledge dataset MMLU-it, "
```

File: src/euroeval/dataset_configs.py [hunk 11]
```diff
@@ -1883,6 +2048,23 @@ def get_dataset_config(dataset_name: str) -> DatasetConfig:
     max_generated_tokens=5,
 )
 
+HELLASWAG_IT_CONFIG = DatasetConfig(
+    name="hellaswag-it",
+    pretty_name="the truncated version of the Italian common-sense reasoning dataset "
+    "HellaSwag-it, translated from the English HellaSwag dataset",
+    huggingface_id="ScandEval/hellaswag-it-mini",
```
Comment (saattrupdan) [inline src/euroeval/dataset_configs.py]:
```suggestion
    huggingface_id="EuroEval/hellaswag-it-mini",
```

File: src/euroeval/dataset_configs.py [hunk 12]
```diff
@@ -1370,6 +1504,20 @@ def get_dataset_config(dataset_name: str) -> DatasetConfig:
     max_generated_tokens=256,
 )
 
+ILPOST_SUM_CONFIG = DatasetConfig(
+    name="ilpost-sum",
+    pretty_name="the truncated version of the Italian summarisation dataset IlPost",
+    huggingface_id="ScandEval/ilpost-sum",
```
Comment (saattrupdan) [inline src/euroeval/dataset_configs.py]:
```suggestion
    huggingface_id="EuroEval/ilpost-sum",
```

File: src/euroeval/dataset_configs.py [hunk 13]
```diff
@@ -1158,6 +1275,23 @@ def get_dataset_config(dataset_name: str) -> DatasetConfig:
     max_generated_tokens=32,
 )
 
+SQUAD_IT_CONFIG = DatasetConfig(
+    name="squad-it",
+    pretty_name="the truncated version of the Italian reading comprehension dataset "
+    "SQuAD-it, translated from the English SQuAD dataset",
+    huggingface_id="ScandEval/squad-it-mini",
```
Comment (saattrupdan) [inline src/euroeval/dataset_configs.py]:
```suggestion
    huggingface_id="EuroEval/squad-it-mini",
```

File: src/euroeval/dataset_configs.py [hunk 14]
```diff
@@ -912,6 +1011,24 @@ def get_dataset_config(dataset_name: str) -> DatasetConfig:
     max_generated_tokens=5,
 )
 
+SCALA_IT_CONFIG = DatasetConfig(
+    name="scala-it",
+    pretty_name="the Italian part of the linguistic acceptability dataset ScaLA",
+    huggingface_id="ScandEval/scala-it",
```
Comment (saattrupdan) [inline src/euroeval/dataset_configs.py]:
```suggestion
    huggingface_id="EuroEval/scala-it",
```

File: src/euroeval/dataset_configs.py [hunk 15]
```diff
@@ -244,6 +244,26 @@ def get_dataset_config(dataset_name: str) -> DatasetConfig:
     max_generated_tokens=5,
 )
 
+SENTIPOLC_CONFIG = DatasetConfig(
+    name="sentipolc16",
+    pretty_name="the truncated version of the Italian sentiment classification "
+    "dataset Sentipolc-16",
+    huggingface_id="ScandEval/sentipolc16-mini",
```
Comment (saattrupdan) [inline src/euroeval/dataset_configs.py]:
```suggestion
    huggingface_id="EuroEval/sentipolc16-mini",
```

File: src/scripts/create_ilpost_sum.py [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+
+    # Collect datasets in a dataset dictionary
+    dataset = DatasetDict(
+        train=Dataset.from_pandas(train_df, split=Split.TRAIN),
+        val=Dataset.from_pandas(val_df, split=Split.VALIDATION),
+        test=Dataset.from_pandas(test_df, split=Split.TEST),
+    )
+
+    # Create dataset ID
+    dataset_id = "ScandEval/ilpost-sum"
```
Comment (saattrupdan) [inline src/scripts/create_ilpost_sum.py]:
```suggestion
    dataset_id = "EuroEval/ilpost-sum"
```

File: src/scripts/create_sentipolc16.py [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+
+    # Collect datasets in a dataset dictionary
+    dataset = DatasetDict(
+        train=Dataset.from_pandas(train_df, split=Split.TRAIN),
+        val=Dataset.from_pandas(val_df, split=Split.VALIDATION),
+        test=Dataset.from_pandas(test_df, split=Split.TEST),
+    )
+
+    # Create dataset ID
+    dataset_id = "ScandEval/sentipolc16-mini"
```
Comment (saattrupdan) [inline src/scripts/create_sentipolc16.py]:
```suggestion
    dataset_id = "EuroEval/sentipolc16-mini"
```

File: src/scripts/create_squad_it.py [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+
+    # Collect datasets in a dataset dictionary
+    dataset = DatasetDict(
+        train=Dataset.from_pandas(train_df, split=Split.TRAIN),
+        val=Dataset.from_pandas(val_df, split=Split.VALIDATION),
+        test=Dataset.from_pandas(test_df, split=Split.TEST),
+    )
+
+    # Create dataset ID
+    mini_dataset_id = "ScandEval/squad-it-mini"
```
Comment (saattrupdan) [inline src/scripts/create_squad_it.py]:
```suggestion
    mini_dataset_id = "EuroEval/squad-it-mini"
```

**Review by saattrupdan — CHANGES_REQUESTED** [review]

File: docs/datasets/italian.md (hunk 5 above)
Comment (viggo-gascou) [reply docs/datasets/italian.md]: Yeah, they provided train/val/test splits and we are just using samples of 1,024/256/2,048 of those splits. In total across the splits there are 181,000 sentences.

File: docs/datasets/italian.md (hunk 4 above)
Comment (viggo-gascou) [reply docs/datasets/italian.md]: They are stored as lists/arrays - this is probably just from how i extracted and printed the samples. I'll make them arrays

File: docs/datasets/italian.md (hunk 6 above)
Comment (viggo-gascou) [reply docs/datasets/italian.md]: Same as for multinerd

File: docs/datasets/italian.md (hunk 7 above)
Comment (viggo-gascou) [reply docs/datasets/italian.md]: Ah nice I could not find the DOI, thanks!

File: docs/datasets/italian.md (hunk 4 above)
Comment (saattrupdan) [reply docs/datasets/italian.md]: Ah right, cool. And thanks.

File: docs/datasets/italian.md (hunk 5 above)
Comment (saattrupdan) [reply docs/datasets/italian.md]: Can you report the original split sizes as well in the docs? Same for the other dataset.

Comment (viggo-gascou) [reply docs/datasets/italian.md]: Yeah, I can do that :)

**Review by saattrupdan — APPROVED** [review]
Looks great now, well done!
